src/main/kotlin/com/github/jengelman/gradle/plugins/shadow/transformers/PropertiesFileTransformer.kt

```diff
@@ -215,6 +223,10 @@ public open class PropertiesFileTransformer @Inject constructor(
   override fun hasTransformedResource(): Boolean = propertiesEntries.isNotEmpty()
 
   override fun modifyOutputStream(os: ZipOutputStream, preserveFileTimestamps: Boolean) {
+    if (conflicts.isNotEmpty()) {
+      throw GradleException("The following properties files have conflicting property values and cannot be merged:${conflicts.joinToString(separator = "\n * ", prefix = "\n * ")}")
```

Comment:
We can improve the error message by counting times like https://github.com/GradleUp/shadow/blob/bd7f137e0123c093302befa1ff204e64c76ede45/src/main/kotlin/com/github/jengelman/gradle/plugins/shadow/tasks/ShadowCopyAction.kt#L131-L134